tasks/main.yml

```diff
@@ -1,28 +1,45 @@
 ---
+# Reboot if required before installing new updates
+- name: Check if reboot is needed
+  ansible.windows.win_powershell:
+    script: |
+      $RebootPending = Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Component Based Servicing\RebootPending"
+      $RebootRequired = $RebootPending `
+      -or (Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootRequired") `
+      -or (Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootPending")
+      if ($RebootRequired) {
+          Write-Output "YES"
+      } else {
+          Write-Output "NO"
+      }
+  register: reboot_needed
 
-#Reboot if required before installing new updates
-- name: If reboot required before checking and installing new updates, reboot the host.
-  win_reboot:
-   reboot_timeout_sec: 1500
-  when: ansible_reboot_pending
+- name: Reboot host before check for update
+  ansible.windows.win_reboot:
+    reboot_timeout_sec: 3600
+  when: reboot_needed.output is defined and "YES" in reboot_needed.output
```

Comment:
By using the `$Ansible.Result` variable the only thing you need here is:

```
when: reboot_needed.result
```

Reply:
Code changed as suggested.